composition-js/src/merging/merge.ts

```diff
@@ -2902,6 +2909,42 @@ class Merger {
     });
   }
 
+  // After merging, if we added any join__directive directives, we want to
+  // remove types imported from the original `@link` directive to avoid
+  // orphaned types. When extractSubgraphsFromSupergraph is called, it will
+  // add the types back to the subgraph.
+  //
+  // TODO: this doesn't handle renamed imports
+  private removeTypesAfterJoinDirectiveSerialization(schema: Schema) {
```

Comment:
I believe this TODO means getting the `name` in `${namespace}__${name}` from the `link.arguments().args.import` argument rather than using `featureDefinition.allElementNames()` (since those are the original names, ignoring potential renaming).

However, I believe all the types we're removing here are the auto-imported types to support the various arguments of `@source` and `@connect`, like `connect__URLPathTemplate` or `connect_HTTPHeaderMapping`, so it's unlikely anyone would explicitly `@link`-import them, so they probably won't (ever?) be renamed, so this TODO is probably moot (or very low priority).